# Examples and Exercises from Think Stats, 2nd Edition
DSC 530 Week 5

Astrid Fuentes

## Exercises

**Exercise 5-1:** In the BRFSS (see Section 5.4), the distribution of heights is roughly normal with parameters µ = 178 cm and σ = 7.7 cm for men, and µ = 163 cm and σ = 7.3 cm for women.

In order to join Blue Man Group, you have to be male between 5’10” and 6’1” (see http://bluemancasting.com). What percentage of the U.S. male population is in this range? Hint: use `scipy.stats.norm.cdf`.

`scipy.stats` contains objects that represent analytic distributions

In [3]:
# import the necesary library
import scipy.stats

For example <tt>scipy.stats.norm</tt> represents a normal distribution.

In [4]:
# Define the parameters for the normal distribution of men heights (µ = 178 cm and σ = 7.7 cm )
mu = 178
sigma = 7.7
# Calculate the distribution using scipy.stats.norm and the given parameters
dist = scipy.stats.norm(loc=mu, scale=sigma)
type(dist)

scipy.stats._distn_infrastructure.rv_frozen

A "frozen random variable" can compute its mean and standard deviation.

In [5]:
# Calculate mean and standard deviation of the distribution
dist.mean(), dist.std()

(178.0, 7.7)

It can also evaluate its CDF.  How many people are more than one standard deviation below the mean?  About 16%

In [14]:
# Calculate the CDF given the two parameters
cdf = dist.cdf(mu-sigma) 
# Show the CDF as a percent rounded to two decimal places
percent = round(cdf *100,2)
percent

15.87

How many people are between 5'10" and 6'1"?

In [19]:
# Calculate the low, high, and diff. 

low = dist.cdf(177.8)    # 5'10" = 177.8 cm
high = dist.cdf(185.4)   # 6'1" = 185.4 cm
diff = high-low
diff*100 # Express as a a percent

34.20946829459531

Aproximatelly 34% of the US Male populaton are between 5'10'' and 6'1'' and could therefor join the Blue Man Group.

**Exercise 5-2:** To get a feel for the Pareto distribution, let’s see how different the world would be if the distribution of human height were Pareto. With the parameters xm = 1 m and α = 1.7, we get a distribution with a reasonable minimum, 1 m, and median, 1.5 m.

Plot this distribution. What is the mean human height in Pareto world? What fraction of the population is shorter than the mean? If there are 7 billion people in Pareto world, how many do we expect to be taller than 1 km? How tall do we expect the tallest person to be?

`scipy.stats.pareto` represents a pareto distribution.  In Pareto world, the distribution of human heights has parameters alpha=1.7 and xmin=1 meter.  So the shortest person is 100 cm and the median is 150.

In [20]:
# Define the parameters of the distribution xm = 1 m and α = 1.7 and the distribution using scipy.stats.pareto
alpha = 1.7
xmin = 1       # meter
dist = scipy.stats.pareto(b=alpha, scale=xmin)
dist.median()

1.5034066538560549

What is the mean height in Pareto world?

In [22]:
# use dist.mean to calculate the mean
dist.mean()

2.428571428571429

What fraction of people are shorter than the mean?

In [23]:
# Calculate the CDF
CDF = dist.cdf(dist.mean())
CDF * 100 # Express as a percent

77.87396975652881

77.9 % of the Pareto World populationn are shorter than the mean.

Out of 7 billion people, how many do we expect to be taller than 1 km?  You could use <tt>dist.cdf</tt> or <tt>dist.sf</tt>.

In [25]:
# Use the Survival Function sf which is the same as 1-cdf.
# Use 1000 which is equivalent to 1km and multiply by the population of 7 billion expressed as 7e9

dist.sf(1000) * 7e9

55602.97643069972

Aproximatelly 55,603 people are expented to be taller than 1km (1000m) in Pareto world. 

How tall do we expect the tallest person to be?

In [47]:
# One way to solve this is to search for a height that we
# expect one person out of 7 billion to exceed.

dist.sf(618000) * 7e9  

1.0009620965722905

The tallest person in Pareto World is expected to be at least 618,000 meters or 618 km tall.

In [26]:
# Another way is to use `ppf`, which evaluates the "percent point function", which
# is the inverse CDF.  So we can compute the height in meters that corresponds to
# the probability (1 - 1/7e9).

dist.ppf(1 - 1/7e9)

618349.6106759505

The answer here is a little bit more accurate. 
The tallest person in Pareto World is expected to be 618,349 meters tall.